portfolio/src/main/webapp/script.js

```diff
@@ -191,24 +191,40 @@ function fetchBlobstoreUrl() {
 function fetchLogin() {
   fetch('/login').then(response => response.json()).then(user => {
     if (user.loggedIn == true) {
+      // If logged in user has no nickname, redirect to nickname setup page.
+      if(user.nickname == "") {
+        window.location.replace(user.logUrl);
+      }
+
       // Create element that welcomes the user and prompts them to log out.
       // "Hello, {username}!"
-      // "Click here to log out."
+      // "To change nickname, click here."
+      // "To log out, click here."
```

Comment:
nit: You probably don't need these comments because they just reiterate what you are doing down below

Reply:
Tim told me to add those comments because it wasn't immediately clear what was being added. Should I keep them?

Reply:
Hm how about just moving the comments down closer to their relevant implementation blocks? That should help with readability!

Reply:
Done!